### PR TITLE
ENH: Add code of conduct document

### PR DESCRIPTION
**Pull request by lassoan**
While reviewing project metrics for a grant application we realized that although we have a common agreement about how to conduct operations in the Slicer community, we do not have a formalized document to describe it.

The Contributor Covenant is a short and clear description that captures well how we operate, so we propose to adopt this as code of conduct document.

The only customization that is made is the contact information: "...reported to the community leaders listed
[here](https://github.com/Slicer/Slicer/blob/master/CONTRIBUTING.md#benevolent-dictators-for-life)
via private message on the [Slicer forum](https://discourse.slicer.org)."

### CODE REVIEW

**Review by pieper — APPROVED** [review]
LGTM 👍 

@jcfr it would be good for you to review too (although I'll bet you agree because I believe you suggested this code for VTK).

Then should we post on discourse before merging?

**Review by fedorov — DISMISSED** [review]

**Comment by lassoan** [comment]
> Then should we post on discourse before merging?

I'm writing up a discourse post now.

**Comment by jcfr** [comment]
Thanks for working on this, it makes perfect sense to me. 

Would it be possible to wait until the end of the week before integrating ? Being one of the person that would help enforce the code (see Enforcement section), that would give me some time for me to run this through our compliance/HR department.

**Comment by lassoan** [comment]
We'll submit the grant application with the link to this document on Wednesday, so merging it on Friday should be no problem. The document can evolve in time, so if there are concerns about any parts then we can push parts of it this week and the rest of it when all the questions are answered.

**Comment by jcfr** [comment]
Thanks Andras

For reference, here is the code of conduct that was integrating in ITK:
https://github.com/InsightSoftwareConsortium/ITK/blob/master/CODE_OF_CONDUCT.md

And the associated reporting guidelines:
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/CodeOfConduct/Reporting.md

I am wondering if we should be more specific regarding the way issues may be reported, documented and discussed.

**Comment by lassoan** [comment]
I've reviewed the ITK code of conduct (as an ISC board member) and it is certainly a very nice and comprehensive document, but the Contributor Covenant has two major advantages:
- shorter: describes essentially the same content in a much shorter format, which increases the chance that people actually read and understand it
- standard: if someone is already familiar with the Contributor Covenant (which is quite likely, as it is used by hundreds if not thousands of projects) then he does not have spend time with reading and understanding a custom, Slicer-specific document. Adding a custom code of conduct document would put unnecessary burden on people (similarly to how we put burden on people by using "Slicer license" instead of a standard license).

**Review by cpinter — APPROVED** [review]

**Review by jamesobutler — DISMISSED** [review]

File: CODE_OF_CONDUCT.md
Comment (jcfr) [inline CODE_OF_CONDUCT.md:123]:
```suggestion
```

File: CODE_OF_CONDUCT.md
Comment (jcfr) [inline CODE_OF_CONDUCT.md:131]:
```suggestion
```

File: CODE_OF_CONDUCT.md
Comment (jcfr) [reply CODE_OF_CONDUCT.md:114]: After discussing the proposed code of conduct with our compliance team @kitware,  without a section describing more clearly when and how to use these guidelines, we suggest to remove this section. It will avoid create expectations that we enforce behavior outside of the project.

**Comment by lassoan** [comment]
These are "guidelines", not laws, not binding contract, and not even called rules, so I don't see the point in removing it. Removing the section would:
- send the message that we actually don't fully support this code of conduct, which places us below the open-source software industry standard
- leave us with yet another non-standard, Slicer-specific document to maintain and explain

If Kitware compliance people cannot recommend any other standard code of conduct document that they found acceptable then, for now, I think we should not have one at all. In the meantime, Kitware, as a responsible open-source software company, should work with Code of Covenant (or other standard code of conduct document) maintainers to address any issues or ambiguities and come up with wording that is acceptable for both Kitware and open-source communities.

**Comment by lassoan** [comment]
@jcfr will add some more details from his latest discussions within Kitware.

**Comment by spujol** [comment]
Having a code of conduct is important. What is the current status of the document?

**Comment by jcfr** [comment]
> Kitware [...] should work with Code of Covenant [...] to address any issues or ambiguities and come up with wording that is acceptable for both Kitware and open-source communities

Improving the covenant code of conduct (and/or other ones) would indeed be sensible. 


In the meantime, the topic has been rebased to account for latest comments. More specifically, the "Enforcement Guidelines" have been removed. This will provide more latitude in the handling of specific complaints and will not create expectations that we enforce behavior outside of the project.

The last step will be remove the reference to `Mozilla` which I think are not relevant here.

**Comment by jcfr** [comment]
Reviewing the [list](https://www.contributor-covenant.org/) of projects reported at `www.contributor-covenant.org`, 23% (15/64) of them include the `Enforcement Guidelines` as described in the Contributor Covenant 2.0 or 2.1.


| Project | Code of conduct | Covenant 2.0 or 2.1 Enforcement Guidelines |
|--|--|--|
| atom | https://github.com/atom/atom/blob/master/CODE_OF_CONDUCT.md | |
| angular | https://github.com/angular/code-of-conduct | |
| babel | https://github.com/babel/babel/blob/main/CODE_OF_CONDUCT.md | |
| bootstrap | https://github.com/twbs/bootstrap/blob/main/CODE_OF_CONDUCT.md | yes |
| rubygems | https://github.com/rubygems/bundler/blob/master/CODE_OF_CONDUCT.md | yes |
| chef | https://docs.chef.io/community_guidelines/ | |
| Cloud Native Computing Foundation | https://github.com/cncf/foundation/blob/main/code-of-conduct.md | |
| CocoaPods | https://github.com/CocoaPods/CocoaPods/blob/master/CODE_OF_CONDUCT.md | |
| Code.gov | https://github.com/GSA/code-gov/blob/master/CODE_OF_CONDUCT.md | |
| creativecommons | https://opensource.creativecommons.org/community/code-of-conduct/ | yes |
| cucumber | https://github.com/cucumber/common/blob/main/CODE_OF_CONDUCT.md | |
| crystal-lang | https://github.com/crystal-lang/crystal/blob/master/CODE_OF_CONDUCT.md | |
| curl | https://github.com/curl/curl/blob/master/docs/CODE_OF_CONDUCT.md | |
| diaspora | https://github.com/diaspora/diaspora/blob/develop/CODE_OF_CONDUCT.md | |
| discourse | https://github.com/discourse/discourse/blob/main/docs/code-of-conduct.md | yes |
| eclipse | https://www.eclipse.org/org/documents/Community_Code_of_Conduct.php | |
| electron | https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md | yes |
| elixir | https://github.com/elixir-lang/elixir/blob/main/CODE_OF_CONDUCT.md | yes |
| exercism | https://github.com/exercism/exercism/blob/main/CODE_OF_CONDUCT.md | |
| flickr | https://blog.flickr.net/en/2022/02/14/safer-internet-day-and-open-source-codes-of-conduct/ | yes |
| gatsbyjs | https://www.gatsbyjs.com/contributing/code-of-conduct/ | |
| git | https://github.com/git/git/blob/master/CODE_OF_CONDUCT.md | yes |
| gitlab | https://about.gitlab.com/community/contribute/code-of-conduct/ | |
| https://go.dev/conduct | |
| Google | https://opensource.google/docs/releasing/template/CODE_OF_CONDUCT/ | |
| Homebrew | https://github.com/Homebrew/.github/blob/master/CODE_OF_CONDUCT.md | |
| Intel | https://01.org/blogs/2018/intel-covenant-code | |
| jekyll | https://github.com/jekyll/jekyll/blob/master/.github/CODE_OF_CONDUCT.markdown | |
| jenkins | https://www.jenkins.io/project/conduct/ | |
| jruby | https://github.com/jruby/jruby/blob/master/CODE_OF_CONDUCT.md | |
| json-schema-org | https://github.com/json-schema-org/.github/blob/main/CODE_OF_CONDUCT.md | yes |
| juneteenthconf | https://juneteenthconf.com/code-of-conduct/ | yes |
| hanamirb | https://hanamirb.org/community/#code-of-conduct | |
| Kong | https://github.com/Kong/kong/blob/master/CODE_OF_CONDUCT.md | |
| Kubernetes | https://github.com/cncf/foundation/blob/main/code-of-conduct.md | |
| Linux | https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8a104f8b5867c682d994ffa7a74093c54469c11f | |
| Mastodon | https://github.com/mastodon/mastodon/blob/main/CODE_OF_CONDUCT.md | |
| Metasploit Framework | https://github.com/rapid7/metasploit-framework/blob/master/CODE_OF_CONDUCT.md | |
| Microsoft | https://opensource.microsoft.com/codeofconduct/ | |
| Mono | https://dotnetfoundation.org/about/code-of-conduct | yes |
| Mozilla | https://www.mozilla.org/en-US/about/governance/policies/participation/ | |
| .NET Foundation| https://dotnetfoundation.org/about/code-of-conduct | yes |
| Node.js | https://github.com/nodejs/admin/blob/HEAD/CODE_OF_CONDUCT.md | yes |
| Rails | https://rubyonrails.org/conduct | |
| rbenv | https://github.com/rbenv/rbenv/blob/master/CODE_OF_CONDUCT.md | |
| React | https://github.com/facebook/react/blob/main/CODE_OF_CONDUCT.md | |
| ROM| https://github.com/rom-rb/rom/blob/main/CODE_OF_CONDUCT.md | | 
| RSpec | https://github.com/rspec/rspec-metagem/blob/main/code_of_conduct.md | |
| ruby-community | https://github.com/ruby-community/ruby-community#contributor-code-of-conduct | |
| rubygems | https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md | |
| RubyGems.org | https://github.com/rubygems/rubygems.org/blob/master/CODE_OF_CONDUCT.md | |
| RVM | https://github.com/rvm/.github/blob/master/CODE_OF_CONDUCT.md | |
| Salesforce OSS | https://github.com/salesforce/oss-template/blob/main/CODE_OF_CONDUCT.md | |
| Shoes | https://github.com/shoes/shoes4/blob/main/CODE_OF_CONDUCT.md |
| Spring | https://github.com/spring-projects/.github/blob/main/CODE_OF_CONDUCT.md | |
| Swift | https://www.swift.org/code-of-conduct/ | |
| Symfony | https://symfony.com/doc/current/contributing/code_of_conduct/code_of_conduct.html | yes |
| Target | https://github.com/target/flottbot/blob/main/.github/CODE_OF_CONDUCT.md | |
| TensorFlow | https://github.com/tensorflow/tensorflow/blob/master/CODE_OF_CONDUCT.md | |
| Travis CI | https://travis-ci.community/t/code-of-conduct/64 | |
| Twilio | https://github.com/twilio/twilio-python/blob/main/CODE_OF_CONDUCT.md | |
| Visual F# | https://dotnetfoundation.org/about/code-of-conduct | yes |
| Vue.js | https://github.com/vuejs/vue/blob/main/.github/CODE_OF_CONDUCT.md | |
| Yarn| https://github.com/yarnpkg/yarn/blob/master/CODE_OF_CONDUCT.md | |

**Comment by rkikinis** [comment]
Hi JC,
In principle I have no problem with this. The specifics are somewhat confusing to me and I would like to discuss.
Best
Ron

> On Jul 13, 2022, at 1:36 AM, Jean-Christophe Fillion-Robin ***@***.***> wrote:
> 
> 
> Reviewing the list of projects reported at www.contributor-covenant.org, 23% (15/64) of them include the Enforcement Guidelines as described in the Contributor Covenant 2.0 or 2.1.
> 
> Project	Code of conduct	Covenant 2.0 or 2.1 Enforcement Guidelines
> atom	https://github.com/atom/atom/blob/master/CODE_OF_CONDUCT.md	
> angular	https://github.com/angular/code-of-conduct	
> babel	https://github.com/babel/babel/blob/main/CODE_OF_CONDUCT.md	
> bootstrap	https://github.com/twbs/bootstrap/blob/main/CODE_OF_CONDUCT.md	yes
> rubygems	https://github.com/rubygems/bundler/blob/master/CODE_OF_CONDUCT.md	yes
> chef	https://docs.chef.io/community_guidelines/	
> Cloud Native Computing Foundation	https://github.com/cncf/foundation/blob/main/code-of-conduct.md	
> CocoaPods	https://github.com/CocoaPods/CocoaPods/blob/master/CODE_OF_CONDUCT.md	
> Code.gov	https://github.com/GSA/code-gov/blob/master/CODE_OF_CONDUCT.md	
> creativecommons	https://opensource.creativecommons.org/community/code-of-conduct/	yes
> cucumber	https://github.com/cucumber/common/blob/main/CODE_OF_CONDUCT.md	
> crystal-lang	https://github.com/crystal-lang/crystal/blob/master/CODE_OF_CONDUCT.md	
> curl	https://github.com/curl/curl/blob/master/docs/CODE_OF_CONDUCT.md	
> diaspora	https://github.com/diaspora/diaspora/blob/develop/CODE_OF_CONDUCT.md	
> discourse	https://github.com/discourse/discourse/blob/main/docs/code-of-conduct.md	yes
> eclipse	https://www.eclipse.org/org/documents/Community_Code_of_Conduct.php	
> electron	https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md	yes
> elixir	https://github.com/elixir-lang/elixir/blob/main/CODE_OF_CONDUCT.md	yes
> exercism	https://github.com/exercism/exercism/blob/main/CODE_OF_CONDUCT.md	
> flickr	https://blog.flickr.net/en/2022/02/14/safer-internet-day-and-open-source-codes-of-conduct/	yes
> gatsbyjs	https://www.gatsbyjs.com/contributing/code-of-conduct/	
> git	https://github.com/git/git/blob/master/CODE_OF_CONDUCT.md	yes
> gitlab	https://about.gitlab.com/community/contribute/code-of-conduct/	
> https://go.dev/conduct		
> Google	https://opensource.google/docs/releasing/template/CODE_OF_CONDUCT/	
> Homebrew	https://github.com/Homebrew/.github/blob/master/CODE_OF_CONDUCT.md	
> Intel	https://01.org/blogs/2018/intel-covenant-code	
> jekyll	https://github.com/jekyll/jekyll/blob/master/.github/CODE_OF_CONDUCT.markdown	
> jenkins	https://www.jenkins.io/project/conduct/	
> jruby	https://github.com/jruby/jruby/blob/master/CODE_OF_CONDUCT.md	
> json-schema-org	https://github.com/json-schema-org/.github/blob/main/CODE_OF_CONDUCT.md	yes
> juneteenthconf	https://juneteenthconf.com/code-of-conduct/	yes
> hanamirb	https://hanamirb.org/community/#code-of-conduct	
> Kong	https://github.com/Kong/kong/blob/master/CODE_OF_CONDUCT.md	
> Kubernetes	https://github.com/cncf/foundation/blob/main/code-of-conduct.md	
> Linux	https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8a104f8b5867c682d994ffa7a74093c54469c11f	
> Mastodon	https://github.com/mastodon/mastodon/blob/main/CODE_OF_CONDUCT.md	
> Metasploit Framework	https://github.com/rapid7/metasploit-framework/blob/master/CODE_OF_CONDUCT.md	
> Microsoft	https://opensource.microsoft.com/codeofconduct/	
> Mono	https://dotnetfoundation.org/about/code-of-conduct	yes
> Mozilla	https://www.mozilla.org/en-US/about/governance/policies/participation/	
> .NET Foundation	https://dotnetfoundation.org/about/code-of-conduct	yes
> Node.js	https://github.com/nodejs/admin/blob/HEAD/CODE_OF_CONDUCT.md	yes
> Rails	https://rubyonrails.org/conduct	
> rbenv	https://github.com/rbenv/rbenv/blob/master/CODE_OF_CONDUCT.md	
> React	https://github.com/facebook/react/blob/main/CODE_OF_CONDUCT.md	
> ROM	https://github.com/rom-rb/rom/blob/main/CODE_OF_CONDUCT.md	
> RSpec	https://github.com/rspec/rspec-metagem/blob/main/code_of_conduct.md	
> ruby-community	https://github.com/ruby-community/ruby-community#contributor-code-of-conduct	
> rubygems	https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md	
> RubyGems.org	https://github.com/rubygems/rubygems.org/blob/master/CODE_OF_CONDUCT.md	
> RVM	https://github.com/rvm/.github/blob/master/CODE_OF_CONDUCT.md	
> Salesforce OSS	https://github.com/salesforce/oss-template/blob/main/CODE_OF_CONDUCT.md	
> Shoes	https://github.com/shoes/shoes4/blob/main/CODE_OF_CONDUCT.md	
> Spring	https://github.com/spring-projects/.github/blob/main/CODE_OF_CONDUCT.md	
> Swift	https://www.swift.org/code-of-conduct/	
> Symfony	https://symfony.com/doc/current/contributing/code_of_conduct/code_of_conduct.html	yes
> Target	https://github.com/target/flottbot/blob/main/.github/CODE_OF_CONDUCT.md	
> TensorFlow	https://github.com/tensorflow/tensorflow/blob/master/CODE_OF_CONDUCT.md	
> Travis CI	https://travis-ci.community/t/code-of-conduct/64	
> Twilio	https://github.com/twilio/twilio-python/blob/main/CODE_OF_CONDUCT.md	
> Visual F#	https://dotnetfoundation.org/about/code-of-conduct	yes
> Vue.js	https://github.com/vuejs/vue/blob/main/.github/CODE_OF_CONDUCT.md	
> Yarn	https://github.com/yarnpkg/yarn/blob/master/CODE_OF_CONDUCT.md	
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because your review was requested.
> 



—
Ron Kikinis
***@***.***
--------
This email is intended for non-work related messages

**Review by jamesobutler — APPROVED** [review]

**Comment by lassoan** [comment]
Thank you @jcfr, this looks good to me. Github does not allow me to "Approve" the pull request because I created it.

**Review by pieper — APPROVED** [review]
LGTM 👍 Thank you all 🙏

**Review by jcfr — APPROVED** [review]

**Comment by sobieavila** [comment]
This looks good to me. To be clear, Kitware is 100% behind creating a community environment that is safe/welcoming to everyone. The challenge with the enforcement section is in the implied obligation to consider activity outside of the community platforms, without any support of a body of experts who can actually handle those situations. We should not put language into a code of conduct that is intended just to give the impression that we take this seriously, it should all be something that we'll truly stand behind (so while they are just guidelines, they are fairly meaningless if we don't always follow them).

**Pull request merged**
